x-pack/plugins/lens/public/shared_components/axis/extent/axis_extent_settings.tsx

```diff
@@ -134,6 +97,31 @@ export function AxisBoundsControl({
           }}
         />
       ) : null}
+      {canShowNiceValues ? (
+        <EuiFormRow
+          label={i18n.translate('xpack.lens.fullExtent.niceValues', {
+            defaultMessage: 'Round to nice values',
+          })}
+          display="columnCompressedSwitch"
+          fullWidth
+        >
+          <EuiSwitch
+            showLabel={false}
+            label={i18n.translate('xpack.lens.fullExtent.niceValues', {
+              defaultMessage: 'Round to nice values',
+            })}
+            data-test-subj={`${testSubjPrefix}_axisExtent_niceValues`}
+            checked={extent.niceValues == null || extent.niceValues}
+            onChange={({ target: { checked: niceValues } }) => {
+              setExtent({
+                ...extent,
+                niceValues,
+              });
+            }}
```

Comment:
This `niceValues` switch was using the local variable instead of the cleaner event value.

Also this was attempting to set the `mode` to `modeForNiceValues` when the nice switch changed, but since it was hidden for non-allowed `modes` this was superfluous.